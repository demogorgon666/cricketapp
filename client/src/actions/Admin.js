import {
  GET_PLAYERS,
  CREATE_PLAYER,
  CREATE_TEAM,
  EDIT_PLAYER,
  EDIT_TEAM,
  DELETE_PLAYER,
  DELETE_TEAM,
  GET_PLAYER_SEARCH,
  GET_TEAM_SEARCH,
  GET_ALLTEAMS,
  ERROR_TYPE
} from "./Types";
import axios from "axios";

// export const getPlayers = () => dispatch => {
// 	return axios
// 		.get("http://localhost:5000/api/cricketalpha/allPlayer")
// 		.then(res => {
// 			console.log("From Player Actions", res.data.data);
// 			dispatch({
// 				type: GET_PLAYERS,
// 				payload: res.data.data
// 			});
// 		})
// 		.catch(err => {
// 			console.log(err);
// 		});
// };
//
export const getPlayers = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

// export const getTeams = () => dispatch => {
//     return axios
//       .get("http://localhost:5000/api/cricketalpha/allteams")
//       .then(res => {
//         dispatch({
//           type: GET_TEAMS,
//           payload: res.data.data
//         });
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

export const getAllTeams = () => dispatch => {
  console.log(localStorage.getItem("token"));
  return axios
    .get("http://localhost:5000/apis/admin/allteam", {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: GET_ALLTEAMS,
        payload: res.data.data
      });
      console.log(res.data);
    })
    .catch(err => {
      console.log(err);
    });
};

//   export const createPlayer = player => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/player/new", player)
//       .then(res => {
//         dispatch({
//           type: CREATE_PLAYER
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };

//   export const createTeam = team => dispatch => {
//     return axios
//       .post("http://localhost:5000/api/cricketalpha/user/new", team)
//       .then(res => {
//         dispatch({
//           type: CREATE_TEAM
//         });
//         console.log("User created successfully");
//       })
//       .catch(err => {
//         console.log(err);
//       });
//   };
export const createTeam = team => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/team/new", team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_TEAM
      });
      alert("Added one team successfully");
      dispatch(getAllTeams());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const createPlayer = player => dispatch => {
  return axios
    .post("http://localhost:5000/apis/admin/player/new", player, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: CREATE_PLAYER
      });
      alert("Added one player successfully");
      dispatch(getPlayers());
    })
    .catch(err => {
      console.log(err);
      dispatch({ type: ERROR_TYPE, payload: err.response.data.message });
      console.log(err.response.data.message);
    });
};

export const editPlayer = player => dispatch => {
  return axios
    .put(
      "http://localhost:5000/apis/admin/editplayer/" + player.player_id,
      player,
      {
        headers: {
          Authorization: `Bearer ${localStorage.getItem("token")}`
        }
      }
    )
    .then(res => {
      dispatch({
        type: EDIT_PLAYER
      });

      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const editTeam = team => dispatch => {
  return axios
    .put("http://localhost:5000/apis/admin/editteam/" + team.team_id, team, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: EDIT_TEAM
      });
      // dispatch(getUsers());
      alert("Edited Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deletePlayer = player_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteplayer/" + player_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_PLAYER
      });
      dispatch(getPlayers());
      // alert("Player Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const deleteTeam = team_id => dispatch => {
  return axios
    .delete("http://localhost:5000/apis/admin/deleteteam/" + team_id, {
      headers: {
        Authorization: `Bearer ${localStorage.getItem("token")}`
      }
    })
    .then(res => {
      dispatch({
        type: DELETE_TEAM
      });
      dispatch(getAllTeams());
      // alert("Team Deleted Successfully");
    })
    .catch(err => {
      console.log(err);
    });
};

export const getPlayerSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/search/" + searchString)
    .then(res => {
      dispatch({
        type: GET_PLAYER_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTeamSearch = searchString => dispatch => {
  return axios
    .get("http://localhost:5000/apis/admin/searchteam/" + searchString)
    .then(res => {
      dispatch({
        type: GET_TEAM_SEARCH,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
