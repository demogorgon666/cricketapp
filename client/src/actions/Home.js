import {
  GET_PLAYERS,
  GET_STATISTICS,
  GET_PLAYER_STATS,
  GET_TOP_BATSMAN,
  GET_NEWS,
  GET_HOME_RECENT_MATCHES
} from './Types'
import axios from 'axios'

export const getNews = () => dispatch => {
  axios
    .get('http://localhost:5000/apis/news')
    .then(res => {
      dispatch({
        type: GET_NEWS,
        payload: res.data.data
      })
      console.log("from 'then' get news")
    })

    .catch(err => console.log(err))
}

// fetching recent matches
export const getRecentMatches = () => dispatch => {
  console.log('from actions of home recent matches')
  axios
    .get('http://localhost:5000/apis/recentMatches')
    .then(res => {
      dispatch({
        type: GET_HOME_RECENT_MATCHES,
        payload: res.data.data
      })
      console.log("from 'then' get RECENT MATCHES")
    })

    .catch(err => console.log(err))
}
