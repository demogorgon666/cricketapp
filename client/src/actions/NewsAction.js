import { GET_NEWS_ID } from "./Types";
import axios from "axios";
import { id } from "date-fns/esm/locale";

export const getNewsById = id => dispatch => {
  console.log("getnews by id", id);
  return axios
    .get("http://localhost:5000/apis/News/newsbyid/" + id)
    .then(res => {
      console.log("From newsbyid actions", res.data.data);
      dispatch({
        type: GET_NEWS_ID,
        payload: res.data.data,
      });
    })
    .catch(err => {
      console.log(err);
    });
};
