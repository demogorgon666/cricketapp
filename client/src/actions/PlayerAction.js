import {
  GET_PLAYERS,
  GET_SINGLE_PLAYER,
  GET_ALL_BATSMAN,
  GET_ALL_BOWLERS,
  GET_BATSMAN_STATS,
  GET_ODI_BATSMAN_STATS,
  GET_T20_BATSMAN_STATS,
  GET_ODI_BOWLER_STATS,
  GET_T20_BOWLER_STATS,
  GET_TEST_BOWLER_STATS,
  GET_TOP_SIXES
} from "./Types";
import axios from "axios";

export const getPlayers = () => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/allPlayer")
    .then(res => {
      console.log("From Player Actions", res.data.data);
      dispatch({
        type: GET_PLAYERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getSinglePlayer = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/singlePlayer/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions Single player is: ",
        res.data.data[0].player_name
      );
      dispatch({
        type: GET_SINGLE_PLAYER,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmen = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBatsman", match_type)
    .then(res => {
      // console.log("From Player Actions batsman data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BATSMAN,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBowlers = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopBowlers", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_ALL_BOWLERS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTopSixes = match_type => dispatch => {
  return axios
    .post("http://localhost:5000/apis/PlayerInfo/TopSixes", match_type)
    .then(res => {
      // console.log("From Player Actions bowler data is: ", res.data.data);
      dispatch({
        type: GET_TOP_SIXES,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODI-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions odi Batsman stats is: ", res.data.ODI);
      dispatch({
        type: GET_ODI_BATSMAN_STATS,
        payload: res.data.ODI
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getODIBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/ODIBowlerStats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions odi BOWLER stats is: ",
        res.data.ODIBowler
      );
      dispatch({
        type: GET_ODI_BOWLER_STATS,
        payload: res.data.ODIBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBatsmanStats = player_id => dispatch => {
  return axios
    .get(
      "http://localhost:5000/apis/PlayerInfo/Test-Batsman-Stats/" + player_id
    )
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.Test);
      dispatch({
        type: GET_BATSMAN_STATS,
        payload: res.data.Test
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getTestBowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Test-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions test Batsman stats is: ",
        res.data.TestBowler
      );
      dispatch({
        type: GET_TEST_BOWLER_STATS,
        payload: res.data.TestBowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Batsman-Stats/" + player_id)
    .then(res => {
      // console.log("From Player Actions test Batsman stats is: ", res.data.T20);
      dispatch({
        type: GET_T20_BATSMAN_STATS,
        payload: res.data.T20
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getT20BowlerStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/T20-Bowler-Stats/" + player_id)
    .then(res => {
      console.log(
        "From Player Actions ODI BOWLER stats is: ",
        res.data.T20Bowler
      );
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.T20Bowler
      });
    })
    .catch(err => {
      console.log(err);
    });
};

export const getBatsmanStats = player_id => dispatch => {
  return axios
    .get("http://localhost:5000/apis/PlayerInfo/Batsman-Stats/" + player_id)
    .then(res => {
      console.log("From Player Actions test Batsman stats is: ", res.data.data);
      dispatch({
        type: GET_T20_BOWLER_STATS,
        payload: res.data.data
      });
    })
    .catch(err => {
      console.log(err);
    });
};
