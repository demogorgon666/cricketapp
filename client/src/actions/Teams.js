import {
	GET_TEAMS,
	GET_RANKS,
	GET_MATCHBYTEAMID,
	GET_TEAM_BATSMEN,
	GET_TEAM_BOWLERS,
	GET_TEAM_FIXTURES,
	GET_TEAM_STATS
} from "./Types";

import { startLoading, stopLoading } from "./LoadingAction";

import axios from "axios";

export const getTeams = match_type => dispatch => {
	return axios
		.post("http://localhost:5000/cricketalpha/teams", match_type)
		.then(res => {
			dispatch({
				type: GET_TEAMS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getRanks = (gender, ranking) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/rankings/" + gender,
			ranking
		)
		.then(res => {
			dispatch({
				type: GET_RANKS,
				payload: res.data.data
			});
			console.log("from actions", ranking);
		})
		.catch(err => {
			console.log(err);
		});
};

export const getMatch = (team_id, gender, match_type) => dispatch => {
	dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/match/" +
				team_id +
				"/" +
				gender,
			match_type
		)
		.then(res => {
			dispatch(stopLoading());
			dispatch({
				type: GET_MATCHBYTEAMID,
				payload: res.data.data
			});
		})
		.catch(err => {
			dispatch(startLoading());
			console.log(err);
		});
};

export const getTeamBatsmen = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbatsmen/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BATSMEN,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamBowlers = (gender, match_type) => dispatch => {
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/topbowlers/" + gender,
			match_type
		)
		.then(res => {
			dispatch({
				type: GET_TEAM_BOWLERS,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

//NIKHIL
export const getFixtures = team_name => dispatch => {
	console.log("get fix from actions", team_name);
	return axios
		.post("http://localhost:5000/cricketalpha/teams/fixtures", team_name)
		.then(res => {
			dispatch({
				type: GET_TEAM_FIXTURES,
				payload: res.data.data
			});
		})
		.catch(err => {
			console.log(err);
		});
};

export const getTeamStats = (team_id, gender, type) => dispatch => {
	console.log("getTeamStats action- ", team_id, gender, type);

	// dispatch(startLoading());
	return axios
		.post(
			"http://localhost:5000/cricketalpha/teams/stats/" +
				team_id +
				"/" +
				gender,
			type
		)

		.then(res => {
			// dispatch(stopLoading());
			console.log("getTeamStats Action");
			dispatch({
				type: GET_TEAM_STATS,
				payload: res.data.data
			});
		})
		.catch(err => {
			// dispatch(startLoading());
			console.log(err);
		});
};
