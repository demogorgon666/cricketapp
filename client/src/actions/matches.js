import { GET_RECENT_MATCHES, GET_MATCHES_DATE, GET_MATCH_DETAILS_BY_ID } from './Types'
import axios from "axios";

const url = 'http://localhost:5000/api/matches';

export const getRecentMatches = (date) => dispatch => {
    return axios
        .get(url + "/recent/" + date, {
        })
        .then(res => {
            dispatch({
                type: GET_RECENT_MATCHES,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getMatchesDate = () => dispatch => {
    return axios
        .get(url + "/bydate", {
        })
        .then(res => {
            dispatch({
                type: GET_MATCHES_DATE,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

export const getmatchdetailbyId = (id) => dispatch => {
    return axios
        .get(url + "/summary/" + id, {
        })
        .then(res => {
            dispatch({
                type: GET_MATCH_DETAILS_BY_ID,
                payload: res.data.data
            });
            console.log(res.data);
        })
        .catch(err => {
            console.log(err);
        });
};

