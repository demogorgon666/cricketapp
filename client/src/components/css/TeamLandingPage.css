* {
  font-family: Montserrat, sans-serif;
}

.container-team {
  margin-top: 80px;
  margin-left: 2%;
  margin-right: 2%;
}

.h1-team {
  color: #2c3e50;
  font-size: 38px;
}

.grid-class-team {
  grid-gap: 10px;
  width: 250px;
  height: 200px;
  padding: 15px;
  border: 1px solid #707070;
  border-radius: 8px;
  margin: 15px;
  display: inline-flex;
  cursor: pointer;
}

.grid-class-team:hover {
  transform: scale(1.1);
}

.img-team-logo {
  width: 100px;
  border-radius: 50%;
}

.info-team {
  width: 97%;
  text-align: center;
  font-size: 27px;
  margin-top: 5%;
}

.grid-container-team {
  display: grid;
  grid-template-columns: auto auto;
}

.country-team {
  width: 70%;
  float: left;
}

.top-team {
  width: 30%;
  float: right;
  /* margin-top: 72px; */
}

.p-team {
  color: #2c3e50;
}

.cards-team {
  display: flex;
  margin: 5px 1000px 15px 15px;
}
.cardtest-team {
  color: #707070;
  background-color: #16a085;
  color: white;
  width: 100%;
  cursor: pointer;
}

.cardtest-team p {
  color: #ffffff;
}

.cardodi-team {
  color: #707070;
  border: 1px solid #16a085;
  width: 100%;
  cursor: pointer;
}

.cardt20-team {
  color: #707070;
  border: 1px solid #16a085;
  width: 100%;
  cursor: pointer;
}

.cardtest-team:hover {
  background-color: #16a085;
  color: white;
}
.cardodi-team:hover {
  background-color: #16a085;
  color: white;
}
.cardt20-team:hover {
  background-color: #16a085;
  color: white;
}

.cardtest-team:hover .p-card {
  color: white;
}
.cardodi-team:hover .p-card {
  color: white;
}
.cardt20-team:hover .p-card {
  color: white;
}

.p-card-team {
  text-align: center;
}

/* ---------------------------------------------------------------- */

@media only screen and (max-width: 1024px) {
  .grid-container-team {
    grid-template-columns: auto;
  }
  .cards-team {
    margin: 0px 370px 15px 10px;
  }
}

@media only screen and (max-width: 768px) {
  .country-team {
    width: 50%;
  }
  .top-team {
    width: 50%;
    float: right;
  }
  .grid-container-team {
    grid-template-columns: auto;
  }
  .cards-team {
    margin: 0px 445px 10px 15px;
  }
  .grid-class-team {
    margin: 12px;
  }
}

@media only screen and (max-width: 576px) {
  .top-team {
    display: none;
  }
  .grid-container-team {
    grid-template-columns: auto;
  }
  .cards-team {
    margin: 0px 15px 15px 10px;
  }
  .grid-class-team {
    margin: 12px;
  }
}

/* ----------------------------------------------------------------- */

/* .grid-class-topteam {
  width: 350px;
  border: 1px solid #707070;
  border-radius: 8px;
  margin: 15px;
  margin-left: 0px;
  position: fixed;
}

.p-topteam {
  width: 98%;
  text-align: center;
  font-size: 28px;
  color: #2c3e50;
}

.cards {
  display: flex;
  margin-left: 20%;
  margin-right: 20%;
  margin-bottom: 5%;
}
.cardtest {
  color: #707070;
  background-color: #16a085;
  color: white;
  width: 100%;
  cursor: pointer;
}

.cardtest p {
  color: #ffffff;
}

.cardodi {
  color: #707070;
  border: 1px solid #16a085;
  width: 100%;
  cursor: pointer;
}

.cardt20 {
  color: #707070;
  border: 1px solid #16a085;
  width: 100%;
  cursor: pointer;
}

.cardtest:hover {
  background-color: #16a085;
  color: white;
}
.cardodi:hover {
  background-color: #16a085;
  color: white;
}
.cardt20:hover {
  background-color: #16a085;
  color: white;
}

.cardtest:hover .p-card {
  color: white;
}
.cardodi:hover .p-card {
  color: white;
}
.cardt20:hover .p-card {
  color: white;
}

.p-card {
  text-align: center;
}

.list {
  margin-left: 5%;
  display: flex;
  align-items: center;
}
.img-card {
  width: 60px;
  height: 60px;
  margin-right: 5%;
  border-radius: 50%;
}
.p-team-name {
  font-size: 20px;
  margin-right: auto;
}

.h2-team-position {
  font-size: 20px;
  margin-left: auto;
  margin-right: 18px;
}

.hr-team-card {
  margin-left: 5%;
  margin-right: 5%;
} */
