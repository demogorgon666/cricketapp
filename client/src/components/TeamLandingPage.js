import React, { Component } from "react";
import { connect } from "react-redux";
import "./css/TeamLandingPage.css";
import logo from "./images/indialogo.jpg";
import Navbar from "../components/common/Navbar";
import { getTeams } from "../actions/Teams";
import MostWins from "./MostWins";

export class TeamLandingPage extends Component {
  state = {
    testClick: true,
    odiClick: false,
    t20Click: false,
    iplClick: false
  };
  componentDidMount() {
    console.log(this.props);
    // this.props.getTeams();
    let type = {
      match_type: "Test",
      competition: "others"
    };
    this.props.getTeams(type);
  }

  onClickTest = () => {
    this.setState({
      testClick: true,
      odiClick: false,
      t20Click: false,
      iplClick: false
    });
    let type = {
      match_type: "Test",
      competition: "others"
    };
    this.props.getTeams(type);
  };

  onClickOdi = () => {
    this.setState({
      testClick: false,
      odiClick: true,
      t20Click: false,
      iplClick: false
    });
    let type = {
      match_type: "ODI",
      competition: "others"
    };
    this.props.getTeams(type);
  };

  onClickT20 = () => {
    this.setState({
      testClick: false,
      odiClick: false,
      t20Click: true,
      iplClick: false
    });
    let type = {
      match_type: "T20",
      competition: "others"
    };
    this.props.getTeams(type);
  };

  onClickIpl = () => {
    this.setState({
      testClick: false,
      odiClick: false,
      t20Click: false,
      iplClick: true
    });
    let type = {
      match_type: "T20",
      competition: "IPL"
    };
    this.props.getTeams(type);
  };

  render() {
    console.log(this.props);
    return (
      <div>
        <Navbar
          gender={this.props.gender}
          changeGender={getGender => this.props.changeGender(getGender)}
        />
        <div className="container-team">
          <h2 className="h1-team">Teams</h2>
          <div className="centered-team">
            <div className="cards-team">
              <div
                style={{ borderRadius: "8px 0px 0px 8px" }}
                className={this.state.testClick ? "cardtest" : "cardodi"}
                onClick={this.onClickTest}
              >
                <p className="p-card">Test</p>
              </div>
              <div
                className={this.state.odiClick ? "cardtest" : "cardodi"}
                onClick={this.onClickOdi}
              >
                <p className="p-card">ODI</p>
              </div>
              <div
                className={this.state.t20Click ? "cardtest" : "cardt20"}
                onClick={this.onClickT20}
              >
                <p className="p-card">IT20</p>
              </div>
              <div
                style={{ borderRadius: "0px 8px 8px 0px" }}
                className={this.state.iplClick ? "cardtest" : "cardt20"}
                onClick={this.onClickIpl}
              >
                <p className="p-card">IPL</p>
              </div>
            </div>
          </div>
          <div className="grid-container-team">
            <div>
              <div className="country-team">
                {this.props.teams.map(teams => (
                  <div
                    className="grid-class-team"
                    onClick={() => {
                      this.props.history.push("/teamdetails/" + teams.team_id, {
                        teams
                      });
                    }}
                  >
                    <div className="info-team">
                      {/* <img src={logo} className="img-team-logo" /> */}
                      <img
                        id="pImage"
                        className="playerImage"
                        // className="img-team-logo"
                        src={`data:image/jpeg;base64,${teams.team_image}`}
                      ></img>
                      <p className="p-team">{teams.team_name}</p>
                    </div>
                  </div>
                ))}
              </div>
              <div className="top-team">
                <MostWins
                  gender={this.props.gender}
                  changeGender={getGender => this.props.changeGender(getGender)}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateTostate = state => ({
  teams: state.TeamsReducer.teams
});

export default connect(
  mapStateTostate,
  { getTeams }
)(TeamLandingPage);
