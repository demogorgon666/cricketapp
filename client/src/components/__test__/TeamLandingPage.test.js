import React from "react";
import { shallow } from "enzyme";
import { TeamLandingPage } from "../TeamLandingPage";
import { getTeams } from "../../actions/Teams";

var teams = [];
const wrapper = shallow(<TeamLandingPage getTeams={getTeams} teams={teams} />);

describe("Test TeamLandingPage Component", () => {
  it("render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should have links for Test,ODI,T20 and IPL", () => {
    expect(
      wrapper
        .find("p")
        .at(0)
        .text()
    ).toBe("Test");

    expect(
      wrapper
        .find("p")
        .at(1)
        .text()
    ).toBe("ODI");

    expect(
      wrapper
        .find("p")
        .at(2)
        .text()
    ).toBe("IT20");

    expect(
      wrapper
        .find("p")
        .at(3)
        .text()
    ).toBe("IPL");
  });
  it("should change the state of testClicked to true when test tab is clicked", () => {
    const test = wrapper.find("div").at(4);
    test.simulate("click");
    expect(wrapper.state().testClick).toBe(true);
    expect(wrapper.state().odiClick).toBe(false);
    expect(wrapper.state().t20Click).toBe(false);
    expect(wrapper.state().iplClick).toBe(false);
  });

  it("should change the state of odiclicked to true when odi tab is clicked", () => {
    const odi = wrapper.find("div").at(5);
    odi.simulate("click");
    expect(wrapper.state().testClick).toBe(false);
    expect(wrapper.state().odiClick).toBe(true);
    expect(wrapper.state().t20Click).toBe(false);
    expect(wrapper.state().iplClick).toBe(false);
  });

  it("should change the state of t20Click to true when IT20 tab is clicked", () => {
    const t20 = wrapper.find("div").at(6);
    t20.simulate("click");
    expect(wrapper.state().odiClick).toBe(false);
    expect(wrapper.state().testClick).toBe(false);
    expect(wrapper.state().t20Click).toBe(true);
    expect(wrapper.state().iplClick).toBe(false);
  });
  it("should change the state of iplClick to true when IPL tab is clicked", () => {
    const ipl = wrapper.find("div").at(7);
    ipl.simulate("click");
    expect(wrapper.state().odiClick).toBe(false);
    expect(wrapper.state().testClick).toBe(false);
    expect(wrapper.state().t20Click).toBe(false);
    expect(wrapper.state().iplClick).toBe(true);
  });
  it("should check if the tabs are getting clicked ", () => {
    wrapper
      .find("div")
      .at(0)
      .simulate("click");
  });
  // it("history.push", () => {
  //   const spy = jest.spyOn(Component.prototype, "getTeams");
  //   const wrapper = mount(<Component {...props} />);
  //   wrapper.instance().getTeams();
  //   expect(spy).toHaveBeenCalled();
  // });
  // expect(spy).toHaveBeenCalled();
  // it("history.push", () => {
  //   const historyMock = { push: jest.fn() };
  //   expect(historyMock.push.mock.calls[0]).toEqual();
  // });
  // it("should have exactly 14 div tags", () => {
  //   expect(wrapper.find("div").length).toBe(14);
  // });
  // it("renders child correctly", () => {
  //   expect(
  //     wrapper
  //       .find(".country-team")
  //       .children()
  //       .find("p")
  //   ).to.have.length(teams.length);
  // });
  // test("click", () => {
  //   wrapper
  //     .find("div.grid-class-team")
  //     .first()
  //     .simulate("click");
  // });
  it("should have no br tags", () => {
    expect(wrapper.find("br").length).toBe(0);
  });
  it("should have className = 'container-team' exacty 1", () => {
    expect(wrapper.find(".container-team").length).toBe(1);
  });
  it("should have className = 'h1-team' exacty 1", () => {
    expect(wrapper.find(".h1-team").length).toBe(1);
    expect(wrapper.find(".h1-team").text()).toBe("Teams");
  });
  it("should have className = 'grid-container-team' exacty 1", () => {
    expect(wrapper.find(".grid-container-team").length).toBe(1);
  });
  it("should have className = 'country-team' exacty 1", () => {
    expect(wrapper.find(".country-team").length).toBe(1);
  });
  // it("should have className = 'grid-class-team' exacty 1", () => {
  //   expect(wrapper.find(".grid-class-team").length).toBe(1);
  // });
  // it("should have className = 'info-team' exacty 1", () => {
  //   expect(wrapper.find(".info-team").length).toBe(4);
  // });
  it("should have className = 'top-team' exacty 1", () => {
    expect(wrapper.find(".top-team").length).toBe(1);
  });
  it("should have className = 'top-team' exacty 1", () => {
    expect(wrapper.find(".top-team").length).toBe(1);
  });
  it("renders", () => {
    expect(wrapper.exists()).toBe(true);
  });
  // -------------------------
  // it("should have className = 'grid-class-topteam' exacty 1", () => {
  //   expect(wrapper.find(".grid-class-topteam").length).toBe(1);
  // });
  // it("should have className = 'cards' exacty 1", () => {
  //   expect(wrapper.find(".cards").length).toBe(1);
  // });
  // it("should have className = 'cardtest' exacty 1", () => {
  //   expect(wrapper.find(".cardtest").length).toBe(1);
  // });
  // it("should have className = 'cardodi' exacty 1", () => {
  //   expect(wrapper.find(".cardodi").length).toBe(1);
  // });
  // it("should have className = 'cardt20' exacty 1", () => {
  //   expect(wrapper.find(".cardt20").length).toBe(1);
  // });
  // it("should have className = 'p-card' exacty 3", () => {
  //   expect(wrapper.find(".p-card").length).toBe(3);
  // });
  // ---------------------------
  // it("should have className = 'list' exacty 4", () => {
  //   expect(wrapper.find(".list").length).toBe(4);
  // });
  // it("should have className = 'img-card' exacty 4", () => {
  //   expect(wrapper.find(".img-card").length).toBe(4);
  // });
  // it("should have className = 'p-team-name' exacty 4", () => {
  //   expect(wrapper.find(".p-team-name").length).toBe(4);
  // });
  // it("should have className = 'h2-team-position' exacty 4", () => {
  //   expect(wrapper.find(".h2-team-position").length).toBe(4);
  // });
  // it("should have className = 'hr-team-card' exacty 4", () => {
  //   expect(wrapper.find(".hr-team-card").length).toBe(4);
  // });
});
