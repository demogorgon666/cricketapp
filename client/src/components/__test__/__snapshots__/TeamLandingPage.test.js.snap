// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TeamLandingPage Component render the component 1`] = `
<div>
  <Connect(Navbar)
    changeGender={[Function]}
    showGender={true}
  />
  <div
    className="container-team"
  >
    <h2
      className="h1-team"
    >
      Teams
    </h2>
    <div
      className="centered-team"
    >
      <div
        className="cards-team"
      >
        <div
          className="cardtest"
          onClick={[Function]}
          style={
            Object {
              "borderRadius": "8px 0px 0px 8px",
            }
          }
        >
          <p
            className="p-card"
          >
            Test
          </p>
        </div>
        <div
          className="cardodi"
          onClick={[Function]}
        >
          <p
            className="p-card"
          >
            ODI
          </p>
        </div>
        <div
          className="cardt20"
          onClick={[Function]}
        >
          <p
            className="p-card"
          >
            IT20
          </p>
        </div>
        <div
          className="cardt20"
          onClick={[Function]}
          style={
            Object {
              "borderRadius": "0px 8px 8px 0px",
            }
          }
        >
          <p
            className="p-card"
          >
            IPL
          </p>
        </div>
      </div>
    </div>
    <div
      className="grid-container-team"
    >
      <div>
        <div
          className="country-team"
        />
        <div
          className="top-team"
        >
          <Connect(MostWins)
            changeGender={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
